model.py

```diff
@@ -37,32 +37,33 @@ def __init__(self, config):
             torch.zeros(config.l_q, 1, 3, 3), requires_grad=True)
         self.sm = nn.Softmax(dim=1)
 
-    def forward(self, X, S1, S2, config):
-        h = self.h(X)
-        r = self.r(h)
-        q = self.q(r)
+    def forward(self, input_view, state_x, state_y, k):
+        """
+        :param input_view: (batch_sz, imsize, imsize)
+        :param state_x: (batch_sz,), 0 <= state_x < imsize
+        :param state_y: (batch_sz,), 0 <= state_y < imsize
+        :param k: number of iterations
+        :return: logits and softmaxed logits
+        """
+        h = self.h(input_view)  # intermediate output
+        r = self.r(h)           # reward output
+        q = self.q(r)           # initial Q value from reward for different actions
         v, _ = torch.max(q, dim=1, keepdim=True)
-        for i in range(0, config.k - 1):
+
+        for i in range(k):
```

Comment:
This should be `range(k-1)` to stick with the paper

Reply:
I see! I made a compromising commit by extracting the q evaluation step as a separate function.